From b14d1c1b684014a65b8b7ea0030f7f070a34e6c5 Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Tue, 4 Jul 2023 10:24:11 -0400
Subject: [PATCH] Remove using library directories as runtime library
 directories

The code in calc_libraries_info in distutils/system_info.py converts
library directories to runtime library directories, which seems to
only make sense in an environment when the external libraries are not
coming from standard system locations.  It is actively harmful in the
OE build environment since it results in sysroot paths getting put
into RPATH in the resulting binaries.  Remove that snippet of code so
that module .so's do not trigger the rpath QA sanity checking.

Upstream-Status: Inappropriate

Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 numpy/distutils/system_info.py | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index 656bc37..fb075a1 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -728,19 +728,6 @@ class system_info:
             else:
                 log.info('Library %s was not found. Ignoring' % (lib))
 
-            if r_dirs:
-                i = self.check_libs(r_dirs, [lib])
-                if i is not None:
-                    # Swap library keywords found to runtime_library_dirs
-                    # the libraries are insisting on the user having defined
-                    # them using the library_dirs, and not necessarily by
-                    # runtime_library_dirs
-                    del i['libraries']
-                    i['runtime_library_dirs'] = i.pop('library_dirs')
-                    dict_append(info, **i)
-                else:
-                    log.info('Runtime library %s was not found. Ignoring' % (lib))
-
         return info
 
     def set_info(self, **info):
-- 
2.40.1

